compiler/resolution/src/org/jetbrains/kotlin/resolve/multiplatform/ExpectedActualResolver.kt

```diff
@@ -91,7 +91,7 @@ object ExpectedActualResolver {
                             Substitutor(expectedClass.declaredTypeParameters, container.declaredTypeParameters)
                         }
                         else null
-                    areCompatibleCallables(declaration, actual, parentSubstitutor = substitutor)
+                    areCompatibleCallables(declaration, actual, commonModule, parentSubstitutor = substitutor)
```

Comment:
@sfs Seems this is redundant

Reply:
I got rid of the default argument to `areCompatibleCallables`, since it uses the wrong `ModuleDescriptor` when called on a resolved class descriptor for a type alias.

To avoid more bugs in the future it's probably better to be explicit here.

Reply:
@sfs areCompatibleCallables asumes platformModule in parameter list, but here commonModule is passed.
Could you explain in mode details why default parameter is wrong for typeAliases?

Reply:
@max-kammerer the platformModule parameter is used in `areCompatibleTypes` to compare two types up to expansion of type aliases for expected and actual type constructors. The `commonModule` contains both declarations, since otherwise you wouldn't have found the expected declaration in the first place.

However, if you expand a typealias, the module for the expanded declaration contains only its own dependencies and in particular might not contain the expected/actual declarations at all. This happens in the `annotationsViaActualTypeAliasFromBinary.kt` test case, where the module for `Jnno` doesn't contain either declaration of `Anno`.

Reply:
@sfs Seems here some misunderstooding: platformModule always contains expect declarations, common only common/expect ones. 
There is only one platform module (common one is merged in platform //@udalov) in 'annotationsViaActualTypeAliasFromBinary' test

Reply:
@max-kammerer If I understand you correctly I could simply change `commonModule` to `actual.module`, right?

I'm not sure I understand what you mean though - is there supposed to be only a single module which is the same for all declarations? This is not what I'm seeing; the declaration for `Jnno` points to a different module ("dependencies for test[...]").

Reply:
@sfs 

> commonModule to actual.module, right?
yes, and it's same as default  parameter

Test is a bit tricky: there is module for java annotation which is used as dependency for kotlin module. And this kotlin module is actually a platform module not common!

Reply:
@max-kammerer I switched it back to using a default argument. Should be fine now.